action.yml

```diff
@@ -6,78 +6,173 @@ inputs:
   db-server-name:
     description: The name of the database server to build the database on.
     required: true
+    default: 'localhost'
   db-server-port:
     description: The port that the database server listens on.
     required: false
     default: '1433'
   db-name:
-    description: The name of the database to bubild.
+    description: The name of the database to build.
+    required: true
+  use-integrated-security:
+    description: Use domain integrated security. This only works on windows. If running on a linux runner, set this to false or don't specify a value. If false, a db-username and db-password should be specified. If true, those parameters will be ignored if specified.
```

Comment:
If this value can be omitted, then it shouldn't be required, right?  It seems like `integrated`, `db-user`, and `db-password` can have `required: false`.  We just need to make sure we have good param validation in place.

Reply:
That seems reasonable. If nothing is provided then fall back to integrated security. I'll put that logic in place!

Reply:
I don't think you mentioned it here, but it was mentioned that maybe we should make the `use-integrated-security` flag default to true. I decided against that for a couple of reasons. The first is that I think the most common use case will be not to use integrated security so workflows can run on linux machines. The second is that we wanted to set the default to true so that if the username, password, and `use-integrated-security` flags were all left off, the action would fall back to using integrated security. But the logic we have to do to fallback is the same even if we do that, i.e. `if (use-integrated-security == false && !username && !password)` (forgive the sudo code). So it doesn't much matter what the default is. With both of those reasons in mind I kept the default value as false.